lua/autorun/wallclimb.lua

```diff
@@ -3,343 +3,343 @@ resource.AddFile("sound/ClimbSound2.wav")
-end)
-end
-ply:GetViewModel():SetNoDraw(false)
-end
-end
+                tracedata = {}
+                tracedata.start = ply:EyePos() + (ply:GetForward() * 32)
+                tracedata.endpos = ply:EyePos() + (ply:GetForward() * 128)
+                tracedata.filter = ply
+                local traceAutoHigh = util.TraceLine(tracedata)
+
+                tracedata = {}
+                tracedata.start = ply:EyePos() + (ply:GetForward() * 129) + (ply:GetUp() * 50)
+                tracedata.endpos = ply:EyePos() + (ply:GetForward() * 140) + (ply:GetUp() * -80)
+                tracedata.filter = ply
+                local traceFloorto = util.TraceLine(tracedata)
+
+                if not (traceAutoLow.Hit) and not (traceAutoHigh.Hit) and not (traceAutoHigh2.Hit) and (traceFloorto.Hit) and ply:OnGround() then
+                    ply:ViewPunch(Angle(-3, 0, 0));
+                    ply:SetLocalVelocity(Vector(0, 0, 260) + (ply:EyeAngles():Right() * 0 + ply:GetForward() * 200))
+                end
+            end
+            -- auto jump close end--
+
+            --high--
+            if ply:KeyDown(IN_SPEED) and ply:KeyDown(IN_FORWARD) and ply:OnGround() and not(ply:IsProne()) then
+
+                if not(ply:Crouching()) then
+                    tracedata = {}
+                    tracedata.start = ply:EyePos() + (ply:GetForward() * 1) + (ply:GetUp() * -10)
+                    tracedata.endpos = ply:EyePos() + (ply:GetForward() * 128) + (ply:GetUp() * 30)
+                    tracedata.filter = ply
+                    local traceAimObj = util.TraceLine(tracedata)
+
+                    tracedata = {}
+                    tracedata.start = ply:EyePos() + (ply:GetUp() * 0) + (ply:GetForward() * -30)
+                    tracedata.endpos = ply:EyePos() + (ply:GetUp() * 100) + (ply:GetForward() * 30)
+                    tracedata.filter = ply
+                    local traceHighHigh2 = util.TraceLine(tracedata)
+
+                    tracedata = {}
+                    tracedata.start = ply:EyePos() + (ply:GetUp() * 21) + (ply:GetForward() * 1)
+                    tracedata.endpos = ply:EyePos() + (ply:GetUp() * 60) + (ply:GetForward() * 64)
+                    tracedata.filter = ply
+                    local traceHighLow = util.TraceLine(tracedata)
+
+                    tracedata = {}
+                    tracedata.start = ply:EyePos() + (ply:GetUp() * 61) + (ply:GetForward() * 1)
+                    tracedata.endpos = ply:EyePos() + (ply:GetUp() * 98) + (ply:GetForward() * 64)
+                    tracedata.filter = ply
+                    local traceHighHigh = util.TraceLine(tracedata)
+                    if ply:KeyDown(IN_DUCK) then
+                        return
+                    end
+
+                    local nearPlayer = 0
+                    local entes = ents.FindInSphere(ply:GetPos(), 60)
+                    for k, v in pairs(entes) do
+                        if (v:IsPlayer() or v:IsNPC()) and v ! = ply then

```

Comment:
only 1 error all your `!=` have a space in them. remove the space and you might want to change them to `~=`.

Reply:
I’ll have the opportunity to fix that when I get home on Saturday